plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,123 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	_ "embed"
+	"fmt"
+
+	"github.com/chrisdalke/gomavlib/v3"
+	"github.com/chrisdalke/gomavlib/v3/pkg/dialects/ardupilotmega"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Plugin state
+type Mavlink struct {
+	// Config param
+	FcuURL                 string   `toml:"fcu_url"`
+	SystemID               uint8    `toml:"system_id"`
+	MessageFilter          []string `toml:"message_filter"`
+	StreamRequestEnable    bool     `toml:"stream_request_enable"`
+	StreamRequestFrequency int      `toml:"stream_request_frequency"`
```

Comment:
If we can use an interval here and `config.Duration` as datatype, a more human-readable format is possible like `200ms`...

Reply:
Unfortunately the request frequency has to be in integer frequencies in hertz so this might have some weird edge cases when casting since some of the possible values have floating-point inprecision:

```
1 hz	1 second
2 hz	0.5 seconds
3 hz	0.3333... seconds
4 hz	0.25 seconds
5 hz	0.2 seconds
6 hz	0.1666... seconds
7 hz	0.1429... seconds
8 hz	0.125 seconds
9 hz	0.1111... seconds
10 hz	0.1 seconds
20 hz	0.05 seconds
```

We could throw an error if the period doesn't match an integer frequency, or choose the closest frequency, or keep it the same. let me know!

Reply:
I agree that if you really only can use integer Hertz quantization it's better to keep this in Hertz.